File: Systems-for-Design-and-Implementation/Labs/Lab12/ProiectJava/TransportModel/src/main/java/transport/model/TripDTO.java
package transport.model;

public class TripDTO {
    private String id;
    private String destination;
    private String dateDeparture;
    private String noOfAvailableSeats;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public String getDateDeparture() {
        return dateDeparture;
    }

    public void setDateDeparture(String dateDeparture) {
        this.dateDeparture = dateDeparture;
    }

    public String getNoOfAvailableSeats() {
        return noOfAvailableSeats;
    }

    public void setNoOfAvailableSeats(String noOfAvailableSeats) {
        this.noOfAvailableSeats = noOfAvailableSeats;
    }

    @Override
    public String toString() {
        return "TripDTO:" +
                "id: " + id +
                "; destination: " + destination +
                "; dateDeparture: " + dateDeparture +
                "; noOfAvailableSeats: " + noOfAvailableSeats;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Trip)) return false;
        TripDTO trip = (TripDTO) o;
        return destination.equals(trip.destination) &&
                dateDeparture.equals(trip.dateDeparture) &&
                noOfAvailableSeats.equals(trip.noOfAvailableSeats);
    }

    @Override
    public int hashCode() {
        int result = destination.hashCode();
        result = 31 * result + dateDeparture.hashCode();
        result = 31 * result + noOfAvailableSeats.hashCode();
        return result;
    }
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab11/ProiectCSharp/TransportModel/transport.model/TripDTO.cs
using System;

namespace TransportModel.transport.model
{
    public class TripDTO
    {
        public long id { get; set; }
        public string destination { get; set; }
        public string dateDeparture { get; set; }
        public long noOfAvailableSeats { get; set; }

        public TripDTO()
        {
        }

        public TripDTO(string destination, string dateDeparture, long noOfAvailableSeats)
        {
            this.destination = destination;
            this.dateDeparture = dateDeparture;
            this.noOfAvailableSeats = noOfAvailableSeats;
        }

        public override string ToString()
        {
            return "Trip{" +
                   "destination='" + destination + '\'' +
                   ", dateDeparture=" + dateDeparture +
                   ", noOfAvailableSeats=" + noOfAvailableSeats +
                   '}';
        }
    }
}